adaptors/mqtt/test/e2e/usability/suite_test.go

```diff
@@ -0,0 +1,235 @@
+
+	isOctopusAvailable()
+}
+
+func uninstallOctopus() {
+	// uninstall MQTT adaptor
+	Expect(exec.RunKubectl(nil, GinkgoWriter, "delete", "-f", filepath.Join(testCurrDir, "deploy", "e2e", "all_in_one.yaml"))).
+		Should(Succeed())
+
+	// uninstall octopus
+	Expect(exec.RunKubectl(nil, GinkgoWriter, "delete", "-f", filepath.Join(testRootDir, "deploy", "e2e", "all_in_one.yaml"))).
+		Should(Succeed())
+}
+
+func isOctopusAvailable() {
+	// confirm brain if exist
+	Eventually(func() (bool, error) {
+		var svc corev1.Service
+		var err = k8sCli.Get(testCtx, types.NamespacedName{Namespace: "octopus-system", Name: "octopus-brain"}, &svc)
+		if err != nil {
+			GinkgoT().Log(err)
+			if !apierrs.IsNotFound(err) {
+				return false, err
+			}
+		}
+		if !object.IsActivating(&svc) {
+			return false, nil
+		}
+
+		var deployment appsv1.Deployment
+		err = k8sCli.Get(testCtx, types.NamespacedName{Namespace: "octopus-system", Name: "octopus-brain"}, &deployment)
+		if err != nil {
+			GinkgoT().Log(err)
+			if !apierrs.IsNotFound(err) {
+				return false, err
+			}
+		}
+		if !object.IsActivating(&deployment) {
+			return false, nil
+		}
+
+		return deployment.Status.Replicas > 0 &&
+			deployment.Status.Replicas == deployment.Status.AvailableReplicas, nil
+	}, 300, 1).Should(BeTrue())
+
+	// confirm limb if exist
+	Eventually(func() (bool, error) {
+		var svc corev1.Service
+		var err = k8sCli.Get(testCtx, types.NamespacedName{Namespace: "octopus-system", Name: "octopus-limb"}, &svc)
+		if err != nil {
+			GinkgoT().Log(err)
+			if !apierrs.IsNotFound(err) {
+				return false, err
+			}
+		}
+		if !object.IsActivating(&svc) {
+			return false, nil
+		}
+
+		var daemonset appsv1.DaemonSet
+		err = k8sCli.Get(testCtx, types.NamespacedName{Namespace: "octopus-system", Name: "octopus-limb"}, &daemonset)
+		if err != nil {
+			GinkgoT().Log(err)
+			if !apierrs.IsNotFound(err) {
+				return false, err
+			}
+		}
+		if !object.IsActivating(&daemonset) {
+			return false, nil
+		}
+
+		return daemonset.Status.NumberAvailable > 0 &&
+			daemonset.Status.DesiredNumberScheduled == daemonset.Status.NumberReady, nil
+	}, 300, 1).Should(BeTrue())
+
+}
+
+func deployMQTTServer() {
+	Expect(exec.RunKubectl(nil, GinkgoWriter, "apply", "-f", filepath.Join(testCurrDir, "test", "e2e", "usability", "mosquitto.yaml"))).
+		Should(Succeed())
+
+	isMQTTServerAvailable()
+}
+
+func isMQTTServerAvailable() {
+	GinkgoT().Log("waiting MQTT server for ready")
+	var deploy appsv1.Deployment
+	var key = types.NamespacedName{
+		Name:      "mosquitto",
+		Namespace: "default",
+	}
+	Eventually(func() bool {
+		if err := k8sCli.Get(testCtx, key, &deploy); err != nil {
+			Fail(err.Error())
+		}
+		for _, condition := range deploy.Status.Conditions {
+			if condition.Type == "Available" {
+				return condition.Status == "True"
+			}
+		}
+		return false
+	}, 300, 3).Should(BeTrue())
+}
```

Comment:
To validate a Service Deployment, you can check the available Endpoint of the target Service instead of the available of Deployment.

Reply:
We may have to make sure the deployment is available, not just the endpoint.